crypto/src/main/java/org/hyperledger/besu/crypto/BouncyCastleNodeKey.java

```diff
@@ -41,4 +51,23 @@ public Signature sign(final Bytes32 dataHash) {
   public PublicKey getPublicKey() {
     return nodeKeys.getPublicKey();
   }
+
+  @Override
+  public Bytes32 calculateKeyAgreement(final PublicKey partyKey) {
+    return SECP256K1.calculateKeyAgreement(nodeKeys.getPrivateKey(), partyKey);
+  }
+
+  @Override
+  public Bytes calculateECIESAgreement(final SECP256K1.PublicKey ephPubKey) {
+    final ECDomainParameters dp = SECP256K1.CURVE;
+
+    final CipherParameters pubParam = new ECPublicKeyParameters(ephPubKey.asEcPoint(), dp);
+    final CipherParameters privParam =
+        new ECPrivateKeyParameters(nodeKeys.getPrivateKey().getD(), dp);
+
+    final BasicAgreement agree = new ECDHBasicAgreement();
+    agree.init(privParam);
+    final BigInteger z = agree.calculateAgreement(pubParam);
+    return Bytes.wrap(BigIntegers.asUnsignedByteArray(agree.getFieldSize(), z));
```

Comment:
Would be nicer if both of the key agreement functions were in SECP256k1 class

Reply:
done